### DISCUSSION TITLE
Version 2 Documentation Updates

### DISCUSSION THREAD

**Discussion by jwildfire** [SafetyGraphics/safetyGraphics#610] [General]
I'm working on updates to the documentation for v2 and would like to use this discussion thread for feedback. 

So far, I've rewritten the [introductory "Shiny User Guide" vignette](https://github.com/SafetyGraphics/safetyGraphics/wiki/Vignette:-Shiny-User-Guide). If you've got time to give it a read and provide feedback that would be great. Feel free to edit the wiki page directly with any simple changes or to provide more general feedback as a response on this discussion thread. 

I'll post responses here as more updates are available for review. 

Thanks! 

attn: @xni7 @samussiah @elimillera @JimBuchanan @mli1

**Comment by JimBuchanan** [comment]
Hi Jeremy,

This is nicely done. Just one observation:

This tab is created using the filter functionality included in the {datamods} pacakge<https://github.com/dreamRs/datamods#filter>. Package is misspelled.

One question as an aside…the previous initialization code had an addition for loading larger datasets [safetyGraphicsApp(maxFileSize=100)]. Is there still a file limit that needs a line like this? I do have a very large SAS file of AE data and was wondering if I’m going to choke the app.

**Reply by jwildfire** [reply]
Thanks, Jim. I fixed the typo. 

Good point about the file size. You could load it via `safetyGraphicsApp()`, but It would definitely be an issue in `safetyGraphicsInit()`. I just filed #613 to deal add that `maxFileSize` option there.

**Comment by xni7** [comment]
Vignette looks great, @jwildfire ! Just one comment: it would great to cross-reference various key vignettes, such as the data security one, which is related to deployment and backend data, advanced charting guide for developers etc. Pkgdown site is a nature place to put all relevant articles together. I think it would be really nice to update that prior to the R/Pharma workshop

**Comment by jwildfire** [comment]
Making some progress. Quick updates on the 4 vignettes from the call today: 
- [Overview vignette](https://github.com/SafetyGraphics/safetyGraphics/wiki/Vignette:-Shiny-User-Guide) - Pretty much done. Thanks to @xni7 and @JimBuchanan for the reviews. 
- [Chart Vignette](https://github.com/SafetyGraphics/safetyGraphics/wiki/Vignette:-Chart-Configuration) - Updated up to the Appendices. @elimillera @samussiah @xni7 @rescalan to review as time allows.   
- [Cookbook Vignette](https://github.com/SafetyGraphics/safetyGraphics/wiki/Vignette:-Customization-Cookbook) - Needs some tweaks using the most recent build, but should be pretty easy to update. @jwildfire to work on that after Chart Vignette
- [Technical Vignette](https://github.com/SafetyGraphics/safetyGraphics/wiki/Vignette:-Data-pipelines-and-Security) - Still just an outline. @xni7 to work on updates. We discussed just turning this in to a FAQ for the V2 release.

**Comment by jwildfire** [comment]
@xni7 I just did a quick reformat to the technical vignette to make it a FAQ. 

@samussiah I [added a Q/A](https://github.com/SafetyGraphics/safetyGraphics/wiki/Vignette:-Data-pipelines-and-Security-FAQ#q-can-i-deploy-safetygraphics-to-an-internal-rsconnect-server) about the rsconnect that included the code from earlier. Please Update with any other questions/feedback after you try to deploy internally.

**Reply by xni7** [reply]
not sure what the up arrow means on the left...

**Reply by jwildfire** [reply]
I think it's an upvote button ... though, it's not entirely clear we need both upvotes and the emoji replies ...

**Comment by jwildfire** [comment]
@xni7 @elimillera @samussiah @bzkrouse @rescalan - OK, I think I'm pretty much done with updates in the [charts](https://github.com/SafetyGraphics/safetyGraphics/wiki/Vignette:-Chart-Configuration) vignette. Thoughts/comments/edits very welcome!

**Reply by xni7** [reply]
Thanks @jwildfire, looks very nice! A couple of minor comments

- it may be better to archive those old vignettes from V1 to avoid confusion
- some advanced topics might be needed:
     + after adding hello_world chart, which is transient, how to make it persistent other than re-running the code, e.g. put into a pkg
     + how to connect other pkgs to {safetyGraphics}. e.g. if someone wrote a QTgraphics pkg. perhaps this would be a vignette for {safetyCharts} that can be linked to? e.g. how to develop a custom pkg similar to {safetyCharts}, or contribution guide to {safetyCharts}?

**Reply by jwildfire** [reply]
Thanks @xni7. I updated the sidebar in the wiki to hide the old pages. Also renamed files a bit to match the formatting in the package, so old links might not work. Click "wiki" in the navbar and using the sidebar should work though. 

I like the advanced issues, but might hold off until after v2, unless someone else has time to add them.
